Drivers/PricePartDisplayDriver.cs

```diff
@@ -48,7 +48,7 @@ private Task BuildViewModel(PricePartViewModel model, PricePart part)
 
             model.Price = part.Price;
             model.PriceValue = part.Price.Value;
-            model.PriceCurrency = part.Price.Currency?.CurrencyIsoCode ?? _moneyService.DefaultCurrency.CurrencyIsoCode;
+            model.PriceCurrency = part.Price.Currency?.CurrencyIsoCode == "---" ? _moneyService.DefaultCurrency.CurrencyIsoCode : part.Price.Currency?.CurrencyIsoCode;
```

Comment:
Can you compare with the unspecified currency's symbol explicitly instead of copying its magic string?

Reply:
Comparison changed to be made against UnspecifiedCurrency.